File: src/main/java/me/melijn/jda/commands/fun/SayCommand.java
package me.melijn.jda.commands.fun;

import me.melijn.jda.blub.Category;
import me.melijn.jda.blub.Command;
import me.melijn.jda.blub.CommandEvent;
import me.melijn.jda.utils.ImageUtils;
import net.dv8tion.jda.core.Permission;

import javax.imageio.ImageIO;
import java.awt.*;
import java.awt.image.BufferedImage;
import java.io.File;
import java.io.IOException;

import static me.melijn.jda.Melijn.PREFIX;

public class SayCommand extends Command {

    public SayCommand() {
        this.commandName = "say";
        this.description = "Makes the bot say stuff";
        this.usage = PREFIX + commandName + " [%fontSize%, default is 60] <text>";
        this.aliases = new String[]{"zeg"};
        this.permissions = new Permission[]{Permission.MESSAGE_ATTACH_FILES};
        this.category = Category.FUN;
        this.id = 29;
    }

    @Override
    protected void execute(CommandEvent event) {
        if (event.getGuild() == null || event.hasPerm(event.getMember(), commandName, 0)) {
            if (event.getArgs().isEmpty()) {
                event.sendUsage(this, event);
                return;
            }
            BufferedImage image;
            try {
                String resourcename = event.getExecutor().equalsIgnoreCase("zeg") ? "melijn_zegt.jpg" : "melijn_says.jpg";
                image = ImageIO.read(new File(resourcename));
                String[] args = event.getArgs().split("\\s+");
                Graphics graphics = image.getGraphics();
                Font font = graphics.getFont().deriveFont(60f);
                String text = event.getArgs();

                if (args[0].matches("%\\d+(?:.\\d+)?%")) {
                    try {
                        float fontsize = Float.parseFloat(args[0].substring(1, args[0].length() - 1));
                        font = graphics.getFont().deriveFont(fontsize);
                    } catch (NumberFormatException ignored) {
                        event.reply("The number inside the %% signs must be under 20 figures");
                    }
                    text = text.replaceFirst(args[0] + "\\s+?", "");
                }


                graphics.setFont(font);
                graphics.setColor(Color.DARK_GRAY);

                int startx = 1133;
                int endx = 1800;
                int starty = 82;
                int endy = 1000;

                ImageUtils imageUtils = new ImageUtils();
                image = imageUtils.putText(image, text, startx, endx, starty, endy, graphics);

                event.reply(image);

            } catch (IOException e) {
                e.printStackTrace();
            }

        } else {
            event.reply("You need the permission `" + commandName + "` to execute this command.");
        }
    }
}

File: src/main/kotlin/me/melijn/melijnbot/commands/image/SayCommand.kt
package me.melijn.melijnbot.commands.image

import me.melijn.melijnbot.internals.command.AbstractCommand
import me.melijn.melijnbot.internals.command.CommandCategory
import me.melijn.melijnbot.internals.command.ICommandContext
import me.melijn.melijnbot.internals.utils.ImageUtils
import me.melijn.melijnbot.internals.utils.getIntegerFromArgN
import me.melijn.melijnbot.internals.utils.message.sendRsp
import me.melijn.melijnbot.internals.utils.message.sendSyntax
import me.melijn.melijnbot.internals.utils.removeFirst
import net.dv8tion.jda.api.Permission
import java.awt.Color
import javax.imageio.ImageIO


class SayCommand : AbstractCommand("command.say") {

    init {
        id = 115
        name = "say"
        aliases = arrayOf("zeg")
        cooldown = 2000
        discordChannelPermissions = arrayOf(
            Permission.MESSAGE_ATTACH_FILES
        )
        commandCategory = CommandCategory.IMAGE
    }

    override suspend fun execute(context: ICommandContext) {
        if (context.args.isEmpty()) {
            sendSyntax(context)
        }

        context.initCooldown()
        val fontSize = getIntegerFromArgN(context, 0)
        val input = if (fontSize == null) {
            context.rawArg
        } else {
            context.rawArg.removeFirst("$fontSize").trim()
        }

        val resourceName = if (context.commandParts[1].equals("zeg", true)) {
            "melijn_zegt.jpg"
        } else {
            "melijn_says.jpg"
        }

        val image = SayCommand::class.java.getResourceAsStream("/$resourceName").use { ImageIO.read(it) }
        val graphics = image.graphics
        val font = graphics.font.deriveFont(fontSize?.toFloat() ?: 60f)
        graphics.font = font
        graphics.color = Color.DARK_GRAY
        val startX = 1133
        val endX = 1800
        val startY = 82
        //val endY = 1000
        val endImg = ImageUtils.putText(image, input, startX, endX, startY, graphics)

        sendRsp(context, endImg, "png")
    }
}